File: cpp_codes/src/christmas_tree.cpp
#include <iostream>

void christmas_tree(int rows) {
  for (int i {}; i < rows; ++i){
    for (int j = rows - i; j > 0; j--){
      std::cout << ' ';
    }
    for (int k {}; k < 2 * i + 1; ++k){
      std::cout << '*';
    }
    std::cout << '\n';
  }
}

int main() {
  christmas_tree(10);
  return 0;
}

File: python_codes/src/christmas_tree.py
def christmas_tree(rows):
    for i in range(rows):
        for j in range(rows-i, 0, -1):
            print(' ', rows="")
        for k in range(2*i+1):
            print('*', rows="")
        print()

if __name__ == "__main__":
    christmas_tree(10)